src/steamship/cli/cli.py

```diff
@@ -272,11 +272,17 @@ def serve_local(  # noqa: C901
     client, user, manifest = initialize_and_get_client_and_prep_project()
 
     if workspace:
-        workspace_obj = Workspace.get(client, handle=workspace)
+        # Fetch/create a workspace if one was specified.
+        workspace_obj = Workspace.create(client, handle=workspace, fetch_if_exists=True)
     else:
-        workspace_obj = Workspace.get(client)
+        # Create a new workspace if none was specified.
+        # Otherwise multiple runs co-mingle data in the `default` workspace.
+        workspace_obj = Workspace.create(client)
```

Comment:
Is this namespaced at all to the package/instance that's being used?  It just might be harder to clean up your steamship workspace later without that mapping.

Something like appending the unix ts could help (or a random string suffix that also checks against collision).